flytekit/core/inference.py

```diff
@@ -0,0 +1,182 @@
+"""
+=========
+Inference
+=========
+
+.. currentmodule:: flytekit.core.inference
+
+This module includes inference subclasses that extend the `ModelInferenceTemplate`.
+
+.. autosummary::
+   :nosignatures:
+   :template: custom.rst
+   :toctree: generated/
+
+   NIM
+"""
+
+from typing import Optional
+
+from .utils import ModelInferenceTemplate
+
+
+class NIM(ModelInferenceTemplate):
+    def __init__(
+        self,
+        image: str = "nvcr.io/nim/meta/llama3-8b-instruct:1.0.0",
+        health_endpoint: str = "v1/health/ready",
+        port: int = 8000,
+        cpu: int = 1,
+        gpu: int = 1,
+        mem: str = "20Gi",
+        shm_size: str = "16Gi",
+        # kubernetes secrets
+        ngc_image_secret: Optional[str] = None,
+        ngc_secret_key: Optional[str] = None,
+        ####################
+        env: Optional[dict[str, str]] = None,
+        hf_repo_ids: Optional[list[str]] = None,
+        hf_token_key: Optional[str] = None,
```

Comment:
Is this supposed to be a secret also?

Reply:
This is a Flyte secret. Will update the secrets interface to accept group as well.